website/notifications/emails.py

```diff
@@ -52,25 +52,37 @@ def notify(event, user, node, timestamp, **context):
     return sent_users
 
 def notify_mentions(event, user, node, timestamp, **context):
+    recipient_ids = context.get('new_mentions', [])
+    recipients = OSFUser.objects.filter(guids___id__in=recipient_ids)
+    sent_users = notify_global_event(event, user, node, timestamp, recipients, context=context)
+    return sent_users
+
+def notify_global_event(event, sender_user, node, timestamp, recipients, template=None, context=None):
     event_type = utils.find_subscription_type(event)
     sent_users = []
-    new_mentions = context.get('new_mentions', [])
-    for m in new_mentions:
-        mentioned_user = OSFUser.load(m)
-        subscriptions = get_user_subscriptions(mentioned_user, event_type)
-        for notification_type in subscriptions:
-            if (
-                notification_type != 'none' and
-                subscriptions[notification_type] and
-                m in subscriptions[notification_type]
-            ):
-                store_emails([m], notification_type, 'mentions', user, node,
-                                 timestamp, **context)
-                sent_users.extend([m])
+    users_subscriptions = []
+
+    # Create a list of users' subscription dictionaries
+    for recipient in recipients:
+        users_subscriptions.append(get_user_subscriptions(recipient, event_type))
+
+    # Merge the dictionaries using dict comprehension and exclude 'none' notification type.
+    # Output e.g. {'email_transactional': [u'vsu7t', u'evz43'], 'email_digest': [u'39dbp']}
+    merged_user_subscriptions = {k: [d[k][0] for d in users_subscriptions if d[k]] for k in users_subscriptions[0] if k != 'none'}
+
+    # For each notification type store the list of users
+    for notify_type in merged_user_subscriptions:
+        # Group of users assigned this notification type
+        user_group_type = merged_user_subscriptions[notify_type]
```

Comment:
Iterating through a dictionary, using both keys and values, I think it's better to do:
```py
for notify_type, user_group_type in merged_user_subscriptions.items():
```
instead of looking up each key separately.